### PR TITLE
Solution

### PR DESCRIPTION
**Pull request by Ultramarin**
Solution for task stateful-clones

### CODE REVIEW

File: src/transformStateWithClones.js [hunk 1]
```diff
@@ -8,6 +8,31 @@
  */
 function transformStateWithClones(state, actions) {
   // write code here
+  const result = [];
+  const tempState = { ...state };
+
+  for (const action of actions) {
+    if (action['type'] === 'addProperties') {
```
Comment (YegorKorenkov) [inline src/transformStateWithClones.js:15]: Please use switch..case

File: src/transformStateWithClones.js [hunk 2]
```diff
@@ -8,6 +8,31 @@
  */
 function transformStateWithClones(state, actions) {
   // write code here
+  const result = [];
+  const tempState = { ...state };
+
+  for (const action of actions) {
```
Comment (YegorKorenkov) [inline src/transformStateWithClones.js:14]: You can destruct data from action
const { type, extraData, ... } = action

File: src/transformStateWithClones.js [hunk 3]
```diff
@@ -8,6 +8,31 @@
...
+    if (action['type'] === 'addProperties') {
+      Object.keys(action['extraData']).forEach(key => {
```
Comment (YegorKorenkov) [inline src/transformStateWithClones.js:16]: You can use Object.assign(tempState, extraData) and remove a redundant iteration

**Review by YegorKorenkov — COMMENTED** [review]
Nice work! Please see comments and fix

File: src/transformStateWithClones.js (hunk 3 above)
Comment (Ultramarin) [reply src/transformStateWithClones.js]: I change this

File: src/transformStateWithClones.js (hunk 2 above)
Comment (Ultramarin) [reply src/transformStateWithClones.js:14]: I change this

**Review by tetiana-ivanchenko — CHANGES_REQUESTED** [review]
image

**Comment by Ultramarin** [comment]
> image

I fix this

**Review by VitaliyBondarenko1982 — APPROVED** [review]
Looks good

**Pull request closed without merging**